frontend/packages/knative-plugin/src/utils/kebab-actions.ts

```diff
@@ -21,10 +22,22 @@ const modifyApplicationRefs = [
 ];
 
 export const getKebabActionsForKind = (resourceKind: K8sKind): KebabAction[] => {
+  let menuActions: KebabAction[];
   if (!resourceKind) {
     // no common actions
     return [];
   }
+  const commonActions = _.includes(modifyApplicationRefs, referenceFor(resourceKind))
+    ? [ModifyApplication]
+    : [];
 
-  return _.includes(modifyApplicationRefs, referenceFor(resourceKind)) ? [ModifyApplication] : [];
+  switch (resourceKind.kind) {
+    case ServiceModel.kind:
+      menuActions = [setTrafficDistribution];
+      break;
+    default:
+      menuActions = [];
+      break;
+  }
+  return [...commonActions, ...menuActions];
```

Comment:
You may as well simply `push` into the actions array if the kind is a service. A switch is a bit overkill here.